add 'remove_comments_aggressive' boolean flag (see #11)

girst opened the pull request:
~~Right now, this changes the default behaviour from 'never remove some comments' to 'remove all comments and only preserve some if explicitly said so'. This is in my opinion the approach to take, however, changing the defaults might not be what you want.~~

~~If so, I can rename 'remove_comments_safe' to 'remove_comments_aggressive' and invert its function --it's up to you.~~

edit: github likes this: fix #11

girst committed:
add 'remove_comments_safe' boolean flag

leejo commented:
Thanks! We probably need to maintain backwards compatibility, so I would err on the side of caution and keep the default behaviour and only remove all comments if the option is provided.

girst committed:
s/safe/aggressive/g

girst commented:
thought so and changed it around. this also fixed the ci errors (which was caused by the test not being updated :s)

Re: the test coverage: I've never written tests in perl before; maybe you could have a look at it?

girst committed:
add tests for aggressive comment removal

leejo commented:
I'll have a look sometime this week. Thanks!

leejo commented:
@girst - I've merged into a branch, will look at test failures and coverage. Should have a release out sometime today. Thanks for the contribution!

leejo commented:
@girst - FYI: had load order issues, which is why the test was never passing: https://github.com/leejo/html-packer-perl/commit/daed649fb8d2e7c9b1a93194ceaa682d0d06c0f5

girst commented:
Great! Thanks for fixing the PR ;-)
